dependencies.yaml

```diff
@@ -394,16 +394,12 @@ dependencies:
       - output_types: conda
         packages:
           - *rmm_unsuffixed
-          - pip
-          - pip:
-              - git+https://github.com/python-streamz/streamz.git@master
       - output_types: requirements
         packages:
           # pip recognizes the index as a global option for the requirements.txt file
           # This index is needed for rmm-cu{11,12}.
           - --extra-index-url=https://pypi.nvidia.com
           - --extra-index-url=https://pypi.anaconda.org/rapidsai-wheels-nightly/simple
-          - git+https://github.com/python-streamz/streamz.git@master
     specific:
```

Comment:
I think we can refactor the [`py_rapids_build_cudf` list](https://github.com/rapidsai/cudf/blob/a6b7effebd3d9b2ac7afd7c3ca124b5034b711e4/dependencies.yaml#L89-L98) to include lists named `depends_on_rmm` and `depends_on_pylibcudf` rather than specifying those packages here. We already have a `depends_on_pylibcudf` that we can use, but we should add a `depends_on_rmm` list. It'd help deduplicate a lot.

That would make `build_python_cudf` an empty dependency list, which is probably fine.

Reply:
I totally agree! I'm happy to just push that here as well and use this PR for more `dependencies.yaml` refactoring. It'd help with #15483.

I'll do that and request another review.

Reply:
Ok @bdice I just pushed https://github.com/rapidsai/cudf/pull/16611/commits/8bfbb95cb808b0c28497eff80a056321a159ae3b, which does the following:

* adds a `depends_on_rmm` list
* removes `build_python_pylibcudf` (which only held an `rmm` requirement)
* removes `build_python_cudf` (which only held `rmm` and `pylibcudf` requirements)
* corresponding changes in `files:` section (including some reordering so dependency lists are alphabetical)